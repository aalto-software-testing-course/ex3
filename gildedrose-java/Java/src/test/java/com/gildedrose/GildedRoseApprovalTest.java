package com.gildedrose;

import org.approvaltests.combinations.CombinationApprovals;

import junit.framework.TestCase;

public class GildedRoseApprovalTest extends TestCase {

	public void testSomething() throws Exception {
		String[] items = { "Any" };
		Integer[] sellins = { 0, 10, 42, 100 };
		Integer[] qualities = { 0, 50, 55 };
		CombinationApprovals.verifyAllCombinations(this::checkItem, items, sellins, qualities);
	}

	public Item checkItem(String item, Integer sellin, Integer quality) {
		Item[] items = new Item[] { new Item(item, sellin, quality) };
		GildedRose app = new GildedRose(items);
		app.updateQuality();
		return items[0];
	}

}