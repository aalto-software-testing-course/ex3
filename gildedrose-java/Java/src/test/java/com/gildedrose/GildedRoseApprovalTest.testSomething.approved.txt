[Any, 0, 0] => Any, -1, 0 
[Any, 0, 5] => Any, -1, 3 
[Aged Brie, 0, 0] => Aged Brie, -1, 2 
[Aged Brie, 0, 5] => Aged Brie, -1, 7 
[Sulfuras, Hand of Ragnaros, 0, 0] => Sulfuras, Hand of Ragnaros, 0, 0 
[Sulfuras, Hand of Ragnaros, 0, 5] => Sulfuras, Hand of Ragnaros, 0, 5 
[Backstage passes to a TAFKAL80ETC concert, 0, 0] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
[Backstage passes to a TAFKAL80ETC concert, 0, 5] => Backstage passes to a TAFKAL80ETC concert, -1, 0 
